packages/core/ai_content_platform/vimax/agents/character_portraits.py

```
@@ -0,0 +1,190 @@
+
+        try:
+            portrait = CharacterPortrait(character_name=character.name)
+            total_cost = 0.0
+
+            supported_views = {"front", "side", "back", "three_quarter"}
+            unknown_views = [view for view in self.config.views if view not in supported_views]
+            if unknown_views:
+                raise ValueError(
+                    "Unsupported portrait view(s): "
+                    + ", ".join(sorted(set(unknown_views)))
+                )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)